pkg/controller/migplan/validation.go

```diff
@@ -654,3 +665,79 @@ func containsAccessMode(modeList []kapi.PersistentVolumeAccessMode, accessMode k
 	}
 	return false
 }
+
+type NamespacePods struct {
+	Namespace string         `json:"namespace"`
+	Pods      []PodWithPhase `json:"pods"`
+}
+
+type PodWithPhase struct {
+	Name   string        `json:"name"`
+	Status kapi.PodPhase `json:"status"`
+}
+
+// Validate the pods, all should be healthy befor migration
+func (r ReconcileMigPlan) validatePods(plan *migapi.MigPlan) error {
+	cluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if cluster == nil || !cluster.Status.IsReady() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	failedPodSlice := make([]NamespacePods, 0)
+
+	for _, ns := range plan.Spec.Namespaces {
+		options := k8sclient.InNamespace(ns)
+		podList := kapi.PodList{}
+		err := client.List(context.TODO(), options, &podList)
+		if err != nil {
+			log.Trace(err)
+			return err
+		}
+
+		if len(podList.Items) == 0 {
+			continue
+		}
+
+		podPhaseSlice := make([]PodWithPhase, 0)
+
+		for _, pod := range podList.Items {
+			if pod.Status.Phase == kapi.PodRunning || pod.Status.Phase == kapi.PodSucceeded {
+				continue
+			}
+			podPhaseSlice = append(podPhaseSlice, PodWithPhase{Name: pod.Name, Status: pod.Status.Phase})
+		}
+
+		if len(podPhaseSlice) != 0 {
+			namespacePods := NamespacePods{Namespace: ns, Pods: podPhaseSlice}
+			failedPodSlice = append(failedPodSlice, namespacePods)
+		}
+	}
+
+	podsJSON, err := json.MarshalIndent(failedPodSlice, "", " ")
+	if err != nil {
+		return err
+	}
+
+	if len(failedPodSlice) > 0 {
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     FailedPodStatus,
+			Status:   True,
+			Reason:   NotDone,
+			Category: Warn,
+			Message:  string(podsJSON),
+		})
+	}
```

Comment:
Looking at the associated issue, I see that @jwmatthews  had suggested that we include information on all pods, not just error ones:
"Thinking for first phase we add this for pods in every namespace, we look for their condition and update CR. Was thinking we update with all pod found so that the UI has information so it can display on the validation screen for a MigPlan that ~20 pods are in a good state and one is unhealthy or similar."

This is only warning on unhealthy pods. I'm actually not sure we want to list all pods as suggested, though. Currently the infrastructure is more set up for warning about just the unexpected things. If there are no pods in an error/failed state, then perhaps it's best to just not warn at all, like is done here.

Reply:
@sseago @jwmatthews If total amount of pods in selected namespaces reaches 1000+, then displaying info about all pods is not a good idea.